lib/cim/cim-adapter.ts

```diff
@@ -0,0 +1,49 @@
+import {PimClass} from "../pim/model";
+import {CoreResourceReader} from "../core";
+import {IriProvider} from "./iri-provider";
+
+export interface CimAdapter {
+
+    /**
+     * Resources returned by the adapter are on the PIM level, therefore the PIM IRIs are needed to interconnect them.
+     * Also, the IRIs need to be predictable and revertible so the user is able to get CIM iri from PIM iri.
+     * @param iriProvider
+     */
+    setIriProvider(iriProvider: IriProvider): void;
+
+    /**
+     * Searches the CIM and returns an ordered list of PimClasses.
+     * @param searchQuery Search query
+     * @return List of PimClasses matching the search
+     */
+    search(searchQuery: string): Promise<PimClass[]>;
+
+    /**
+     * Returns a PimClass by CIM iri.
+     * @param cimIri CIM iri of class to be interpreted.
```

Comment:
It is not to be interpreted but to be "get"? I mean it may be interesting to consider alternative name like ```fetch``` or ```load```. ```get``` to me is more like the object is in local memory. 

Also the ```cimIri``` is the IRI of object to return to to be ```interprreted``` right?